lib/features/map_view/controllers/map_controller.dart

```diff
@@ -1,37 +1,43 @@
+import "dart:async";
+
 import "package:flutter/material.dart";
-import "package:flutter_map/flutter_map.dart" as fl_map;
-import "package:riverpod_annotation/riverpod_annotation.dart";
+import "package:flutter_map_animations/flutter_map_animations.dart";
+import "package:flutter_riverpod/flutter_riverpod.dart";
 
 import "../../../config/map_view_config.dart";
 import "bottom_sheet_controller.dart";
 import "controllers_set.dart";
 
-mixin MapController<T extends GoogleNavigable>
-    on AutoDisposeNotifier<fl_map.MapController> {
-  late final MapControllers<T> mapControllers;
+class MyMapController<T extends GoogleNavigable> {
+  final Ref ref;
+  final MapControllers<T> mapControllers;
+  MyMapController(this.ref, this.mapControllers);
 
-  @override
-  fl_map.MapController build() {
-    return fl_map.MapController();
+  // nasty, but no other option I think
+  final _controllerCompleter = Completer<AnimatedMapController>();
```

Comment:
Is this `Completer` here because of build errors? Or could we just create `AnimatedMapController.future`?

Reply:
Completer is just a native dart way to await somewhere until someother part of the code completes with value (in very imperative way with `.complete(value)` - so imo ideal way in this example. But sure, there are few alternative ways to handle this. Riverpod provider could be a solution too, but more bulky